### ISSUE TITLE
with-Python=OFF still leaves python required paths in Help Generation

### ISSUE THREAD

**Issue by kkapanova** [nest/nest-simulator#712]
In order to compile nest without Python, 
I had to comment few lines in nest-2.12.0/doc/CMakeLists.txt and more precisely I removed the following part from the file:
```
if ( NOT CMAKE_CROSSCOMPILING )
  add_custom_target( generate_help ALL )
  # Extract help from all source files in the source code, put them in
  # doc/help and generate a local help index in the build directory containing 
  # links to the help files.
  add_custom_command( TARGET generate_help POST_BUILD
    COMMAND python -B generate_help.py "${PROJECT_SOURCE_DIR}" "${PROJECT_BINARY_DIR}"
    COMMAND python -B generate_helpindex.py "${PROJECT_BINARY_DIR}/doc"
    WORKING_DIRECTORY "${PROJECT_SOURCE_DIR}/extras/help_generator"
    COMMENT "Extracting help information; this may take a litte while."
    )
  # Copy the local doc/help directory to the global installation
  # directory for documentation.
  install( DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/help"
    DESTINATION "${CMAKE_INSTALL_DOCDIR}"
    )
  # Update the global help index to contain all help files that are
  # located in the global installation directory for documentation.
  install( CODE
    "execute_process(
      COMMAND python -B generate_helpindex.py \"${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_DOCDIR}\"
      WORKING_DIRECTORY \"${PROJECT_SOURCE_DIR}/extras/help_generator\"
    )"
    )
endif ()
```
Probably there should and extra if Python statement

**Comment by heplesser** [comment]
@kkapanova I assume that you encountered this problem on a system on which Python was not available at all? I am just trying to understand why this problem did not show up in our CI testing.

This is closely related to #678.

**Comment by kkapanova** [comment]
@heplesser Yes you are correct. No Python was available on the system during the installation. Nevertheless, once this part of the code is removed, the compilation proceeds with no problems..

**Comment by heplesser** [comment]
This will be handled as part of solving #678, therefore I close this issue.

**Comment by heplesser** [comment]
@kkapanova Could you test if the fix proposed by @steffengraber in #715 works on your Python-free system? The easiest way to do this is probably to manually apply the change in #715 to your copy of the NEST sources and then test in a new build directory. Please report if it works or any problems in #715.
